### PR TITLE
magento/magento2#24781: Elasticsearch 6 works with Elasticsearch 5 client on frontend

### PR DESCRIPTION
**Pull request by p-bystritsky**
### Fixed Issues (if relevant)

1. magento/magento2#24781: Elasticsearch 6 works with Elasticsearch 5 client on frontend

### Manual testing scenarios (*)

1. Open PhpStorm, add breakpoints in:
- Elasticsearch 6.x client: https://github.com/magento/magento2/blob/6d6f11b3ff4dc290c1e53c8822e136547b38f05e/app/code/Magento/Elasticsearch6/Model/Client/Elasticsearch.php#L66
- Elasticsearch 5.x client: https://github.com/magento/magento2/blob/6d6f11b3ff4dc290c1e53c8822e136547b38f05e/app/code/Magento/Elasticsearch/Elasticsearch5/Model/Client/Elasticsearch.php#L71
2. Go to Admin >> Stores >> Configurations >> Catalog >> Catalog >> Search
3. Configure Elasticsearch settings to use "Elasticsearch 6.0+" option
4. Click "Test connection", see that code paused at Elasticsearch 6.x client
5. Save all configs, do reindex
6. Go to any category page, see that code paused at Elasticsearch 6.x client
7. Go to any search page, see that code paused at Elasticsearch 6.x client

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @p-bystritsky,

Your changes looks good to me.
Could you cover your changes with integration test?

**Comment by p-bystritsky** [comment]
@ihor-sviziev thank you for review.
I have thought about test but have no idea how to implement it.

**Comment by ihor-sviziev** [comment]
@p-bystritsky i would emulate case that in config we have Elasticsearch 5, then try to get client from `Magento\Elasticsearch\Elasticsearch5\SearchAdapter\ConnectionManager` and check instance of ES5 client, and 2nd test case - with ES6.

What do you think?

**Comment by p-bystritsky** [comment]
@ihor-sviziev test added, please review.

File: dev/tests/integration/testsuite/Magento/Elasticsearch6/SearchAdapter/ConnectionManagerTest.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        $this->connectionManager = $this->objectManager->create(ConnectionManager::class);
+    }
+
+    /**
+     * Test if 'elasticsearch6' search engine returned by connection manager.
+     *
+     * @magentoAppIsolation enabled
+     * @magentoConfigFixture default/catalog/search/engine elasticsearch6
+     */
+    public function testCorrectElasticsearchClient()
```
Comment (ihor-sviziev) [inline dev/tests/integration/testsuite/Magento/Elasticsearch6/SearchAdapter/ConnectionManagerTest.php]: Please add one more test case for ES5, just to make sure that we're not returning ES6 client all the time

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: dev/tests/integration/testsuite/Magento/Elasticsearch6/SearchAdapter/ConnectionManagerTest.php (hunk 1 above)
Comment (ihor-sviziev) [reply dev/tests/integration/testsuite/Magento/Elasticsearch6/SearchAdapter/ConnectionManagerTest.php]: Just updated test myself

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6106](https://jira.corp.magento.com/browse/ENGCOM-6106) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA passed

**Pull request merged**